openedx/adg/lms/courseware_override/helpers.py

```diff
@@ -33,7 +33,9 @@ def get_business_line_prereq_courses(user):
     """
     if not (
         user.is_authenticated and
-        user.application_hub.is_written_application_completed and
+        hasattr(user, 'application_hub') and
+        user.application_hub.is_prerequisite_courses_passed and
+        hasattr(user, 'application') and
         user.application.business_line
```

Comment:
Have we tried `user.is_authenticated and user.application_hub and user.application_hub. is_written_application_completed  .............`
I guess we can replace hasattr with some checks if that exists or not.
Also as this condition is quite complex now, can we please move it to a separate method with a meaningful name?

Reply:
I have tried user.application and user.application_hub but that does not work. Let me add a method in applications model for this.